src/masternodes/mn_rpc.cpp

```diff
@@ -1141,7 +1141,12 @@ UniValue minttokens(const JSONRPCRequest& request) {
                         throw JSONRPCError(RPC_INVALID_REQUEST, "Incorrect Authorization");
                     gotFoundersAuth = true; // we need only one any utxo from founder
                 }
-                rawTx.vin.insert(rawTx.vin.end(), auths.begin(), auths.end());
+
+                for (const auto& auth : auths) {
+                    if (std::find_if(rawTx.vin.begin(), rawTx.vin.end(), [&auth](const CTxIn& txIn) { return txIn == auth; }) == rawTx.vin.end()) {
```

Comment:
Why "find_if" with predicate here instead of just "find"??
   `if (std::find(rawTx.vin.begin(), rawTx.vin.end(), auth) == rawTx.vin.end()) {`
...